src/battle_ai_util.c

```diff
@@ -439,9 +439,9 @@ bool32 IsDamageMoveUnusable(u32 move, u32 battlerAtk, u32 battlerDef)
             return TRUE;
         break;
     case EFFECT_HIT_SET_REMOVE_TERRAIN:
-	if (!(gFieldStatuses & STATUS_FIELD_TERRAIN_ANY))
-	    return TRUE;
-	break;
+        if (!(gFieldStatuses & STATUS_FIELD_TERRAIN_ANY) && (gMovesInfo[move].argument == ARG_TRY_REMOVE_TERRAIN_HIT || gMovesInfo[move].argument == ARG_TRY_REMOVE_TERRAIN_FAIL))
```

Comment:
```suggestion
        if (!(gFieldStatuses & STATUS_FIELD_TERRAIN_ANY) && gMovesInfo[move].argument == ARG_TRY_REMOVE_TERRAIN_FAIL)
```

Ice Spinner doesn't fail if there is no terrain.

Reply:
Sure thing, as soon as I'm back on the computer

Reply:
Done :)